boa_engine/src/builtins/promise/promise_job.rs

```diff
@@ -0,0 +1,194 @@
+use gc::{Finalize, Trace};
+
+use crate::{
+    builtins::promise::{ReactionRecord, ReactionType},
+    job::JobCallback,
+    object::{FunctionBuilder, JsObject},
+    Context, JsValue,
+};
+
+use super::{Promise, PromiseCapability};
+
+pub(crate) struct PromiseJob;
+
+#[derive(Debug, Trace, Finalize)]
+struct ReactionJobCaptures {
+    reaction: ReactionRecord,
+    argument: JsValue,
+}
```

Comment:
This would need some documentation to explain where does it come from, what it's used for, how...